scenarios/cpp/windows/captioning/caption.cpp

```diff
@@ -0,0 +1,580 @@
+		region (region)
+		{}
+};
+
+struct CaptionTime
+{
+	const int startHours;
+	const int endHours;
+	const int startMinutes;
+	const int endMinutes;
+	const float startSeconds;
+	const float endSeconds;
+
+	CaptionTime (
+		int startHours,
+		int endHours,
+		int startMinutes,
+		int endMinutes,
+		float startSeconds,
+		float endSeconds
+	) :
+	startHours (startHours),
+	endHours (endHours),
+	startMinutes (startMinutes),
+	endMinutes (endMinutes),
+	startSeconds (startSeconds),
+	endSeconds (endSeconds)
+	{}
+};
+
+class TooFewArguments : public std::exception
+{
+	public: const char* what () const throw ()
+	{
+		return "Too few arguments.\n";
+	}
+};
+
+/*
+Helper functions
+*/
+
+/* Simple command line argument parsing functions. See
+https://stackoverflow.com/a/868894
+The value parameter to std::find must be std::string. Otherwise std::find will compare the pointer values rather than the string contents.
+*/
+std::optional<std::string> GetCmdOption (char** begin, char** end, const std::string& option)
+{
+	char** result = std::find (begin, end, option);
+	if (result != end)
+	{
+// We found the option (for example, "-o"), so advance from that to the value (for example, "filename").
+		char** value = ++result;
+// std::string (nullptr) is undefined.		
+		if (value != end && nullptr != *value)
+		{
+			return std::optional<std::string>{ std::string (*value) };
+		}
+		else
+		{
+			return std::nullopt;
+		}
+	}
+	else
+	{
+		return std::nullopt;
+	}
+}
+
+inline bool CmdOptionExists (char** begin, char** end, const std::string& option)
+{
+	return std::find (begin, end, option) != end;
+}
+
+std::vector<std::string> split (const std::string& s, char delimiter)
+{
+	std::vector<std::string> tokens;
+	std::string token;
+	std::istringstream tokenStream (s);
+	while (std::getline (tokenStream, token, delimiter))
+	{
+		tokens.push_back (token);
+	}
+	return tokens;
+}
+
+std::string V2EndpointFromRegion (std::string region)
+{
+	return std::string (v2EndpointStart + region + v2EndpointEnd);
+}
+
+CaptionTime CaptionTimeFromTicks (uint64_t startTicks, uint64_t endTicks)
+{
+	const float startSeconds_1 = startTicks / ticksPerSecond;
+	const float endSeconds_1 = endTicks / ticksPerSecond;
+	
+	const int startMinutes_1 = startSeconds_1 / 60;
+	const int endMinutes_1 = endSeconds_1 / 60;	
+	
+	const int startHours = startMinutes_1 / 60;
+	const int endHours = endMinutes_1 / 60;
+	
+	const float startSeconds_2 = fmod (startSeconds_1, 60.0);
+	const float endSeconds_2 = fmod (endSeconds_1, 60.0);
+	
+	const float startMinutes_2 = fmod (startMinutes_1, 60.0);
+	const float endMinutes_2 = fmod (endMinutes_1, 60.0);
+	
+	return CaptionTime (startHours, endHours, startMinutes_2, endMinutes_2, startSeconds_2, endSeconds_2);
+}
+
+std::string AddLanguageToResult (std::shared_ptr<SpeechRecognitionResult> result, UserConfig userConfig)
+{
+	std::ostringstream text;
+	if (true == userConfig.languageIDLanguages.has_value ())
```

Comment:
Can you look at other C++ sample code? I don't think we use the pattern if (true == X). We just do if (X) and if (!X).